sys/dev_vga.c

```diff
@@ -102,10 +103,43 @@ static int dev_vga_lookup(vnode_t *v, const char *name, vnode_t **res) {
   return ENOENT;
 }
 
+static void uiomove_vga_dirent(const char *name, uio_t *uio) {
+  dirent_t *dir = NULL;
+  dir = (dirent_t *)kmalloc(M_VFS, _DIRENT_RECLEN(dir, strlen(name)), M_ZERO);
+  dir->d_namlen = strlen(name);
+  dir->d_reclen = _DIRENT_RECLEN(dir, dir->d_namlen);
+  memcpy(dir->d_name, name, dir->d_namlen + 1);
+  uiomove(dir, dir->d_reclen, uio);
+  kfree(M_VFS, dir);
+}
+
```

Comment:
I have no idea what most of these flags mean and I don't really know how to use them, but notice that:

```
#define DT_CHR 2
#define DT_DIR 4
#define DT_BLK 6
```

Meaning that `DT_DIR & DT_BLK` will give `true` and in case of searching over directories we will want to enter fb and pallete which aren't directories.